operator/src/main/java/oracle/kubernetes/operator/helpers/AnnotationHelper.java

```diff
@@ -25,14 +25,14 @@
    * Marks metadata with annotations that let Prometheus know how to retrieve metrics from the
    * wls-exporter web-app. The specified httpPort should be the listen port of the WebLogic server
    * running in the pod.
-   *
-   * @param meta Metadata
+   *  @param meta Metadata
+   * @param applicationContext the context for the exporter application
    * @param httpPort HTTP listen port
    */
-  static void annotateForPrometheus(V1ObjectMeta meta, int httpPort) {
+  static void annotateForPrometheus(V1ObjectMeta meta, String applicationContext, int httpPort) {
     meta.putAnnotationsItem(
         "prometheus.io/port", "" + httpPort); // should be the ListenPort of the server in the pod
-    meta.putAnnotationsItem("prometheus.io/path", "/wls-exporter/metrics");
+    meta.putAnnotationsItem("prometheus.io/path", applicationContext + "/metrics");
```

Comment:
What is the default `applicationContext`? Unfortunately, these annotations were added at the wrong code point and are present when the model is hashed; therefore, any change to these values causes existing pods to roll.

Reply:
The application context is returned from `ExporterContext.getBasePath()` and depends on whether the annotations are generated for the webapp or the sidecar, and will only change if we swap between those - which is what we want to have happen. For the webapp, it is "/wls-exporter"; for the sidecar, it is "".